### PR TITLE
be: add filename sanitization utility for safe uploads | closes #143

### LINKED ISSUE #143: be: add filename sanitization utility
Implement sanitization logic for uploaded filenames

### tasks
- Create `sanitizeFilename()` in `utils/sanitize.js`
- Convert to lowercase
- Allow ASCII characters only.
- Replace spaces with underscores
- Export and use in upload middleware before saving files

### PR DESCRIPTION
**Pull request by prakhartripath0001**
closes #143 

### why these was needed:
- Unsanitized filenames can cause issues such as:
   - Filesystem write errors (due to invalid characters)
   - Collisions from duplicate or unsafe names
   - Inconsistent naming in uploads and logs

- Added new test suite: `tests-jest/sanitize.test.js`
  - Covers valid, invalid, and edge-case scenarios
  - Ensures consistent behavior and lint compliance

###  How it was tested
- Ran Jest unit tests locally  
  ```bash
  npm test -- tests-jest/sanitize.test.js


<img width="943" height="201" alt="image" src="https://github.com/user-attachments/assets/2866299f-cbe1-43d2-b6c4-8ce474800368" />

### CODE REVIEW

File: server/node-server-app/tests-jest/sanitize.test.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+});
+
+test("sanitizeFilename keeps only one dot in the filename", () => {
+    const result = sanitizeFilename("archive.tar.gz");
+    expect(result).toBe("archive_tar.gz");
+});
+
+test("sanitizeFilename replaces invalid characters with underscores", () => {
+    const result = sanitizeFilename("data@file#1!.txt");
+    expect(result).toBe("data_file_1_.txt");
```
Comment (wtasg) [inline server/node-server-app/tests-jest/sanitize.test.js]: My guess is that the filename part is not getting trimmed before converting spaces to underscores. `data_file_1.txt` should be the output.

File: server/node-server-app/src/utils/sanitize.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * @returns {string} sanitized filename
+ */
+function sanitizeFilename(originalName) {
+    if (!originalName || typeof originalName !== "string") {
+        return "unnamed_file";
+    }
+
+    let name = originalName.toLowerCase();
+
+    name = name.replace(/\s+/g, "_");
```
Comment (wtasg) [inline server/node-server-app/src/utils/sanitize.js]: first convert multiple spaces to single space, then trim the string, then convert single space to underscore.

**Review by wtasg — CHANGES_REQUESTED** [review]

**Comment by praakhartripathi** [comment]
@wtasg , fixed asked changes

File: server/node-server-app/src/utils/sanitize.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    let name = originalName.toLowerCase();
+
+    name = name.replace(/\s+/g, " ");
+    name = name.trim();
+    name = name.replace(/\s/g, "_");
+    name = name.replace(/[^\u0020-\u007E]/g, "");
+
+    const parts = name.split(".");
+    const compoundExtensions = ["tar.gz", "tar.bz2", "tar.xz"];
```
Comment (wtasg) [inline server/node-server-app/src/utils/sanitize.js]: No. Wrong logic.
